### ISSUE TITLE
Should we convert uploaded file metadata into Python types?

### ISSUE THREAD

**Issue by pbugnion** [jupyter-widgets/ipywidgets#2727]
Following PRs #2724 and #2666 , the `FileUpload.value` traitlet looks like:

```python
[{'name': 'example.csv',
  'type': 'text/csv',
  'size': 2548,
  'lastModified': 1536574751152,
  'content': <memory at 0x11d51ab88>}]
```

It is defined as `List(Dict())`.

We could consider the following, in increasing order of complexity:

1. change `lastModified` to be a datetime object. We should check browser compatibility on whether it's always expressed as a timestamp.
2. define a namedtuple or a Python class that exposes the attributes, rather than a dict. This would make the attributes more discoverable.

notifying @jtpio , @jasongrout who have been working on this.

**Comment by jasongrout** [comment]
> define a namedtuple or a Python class that exposes the attributes, rather than a dict. This would make the attributes more discoverable.

Why not have both? Use a Bunch! :)

https://github.com/ipython/traitlets/blob/master/traitlets/utils/bunch.py

(That's what we use for the change argument in traitlets change notifications)

**Comment by jtpio** [comment]
Both suggestions sound good, especially if we want to stay close to the contents api: https://jupyter-notebook.readthedocs.io/en/stable/extending/contents.html#filesystem-entities

**Comment by vidartf** [comment]
As an xref, this might tie in with the datetime serialization in #2715. There, it is serialized along the pattern of `DatePicker`, as a dict of year, month, day, etc. Maybe we should switch to an ISO timestamp everywhere instead?

**Comment by pbugnion** [comment]
I'll try and address this issue.

**Comment by jasongrout** [comment]
Fixed in #2767.
